effectful/handlers/pyro.py

```diff
@@ -330,29 +329,43 @@ def positional_distribution(
 ) -> tuple[TorchDistribution, Naming]:
     shape = d.shape()
     d = defterm(d)
-    dist_constr, args = d.args[0], d.args[1:]
 
-    if not (
-        d.op is call
-        and (
-            issubclass(dist_constr, TorchDistribution)
-            or issubclass(dist_constr, dist.torch_distribution.TorchDistributionMixin)
-        )
-    ):
+    if not isinstance(d, _DistributionTerm):
         raise NotImplementedError
 
     indices = sizesof(d).keys()
     naming = Naming.from_shape(indices, len(shape))
 
     def _to_positional(a):
         if isinstance(a, torch.Tensor):
+            a_indices = sizesof(a)
+            assert len(a_indices) == 0 or set(a_indices.keys()) == set(indices)
```

Comment:
This check implies that all parameter tensors should be broadcasted up to the same named dimensions. Is that actually true? Would something like this fail?
```py
positional_distribution(Normal(locs[i], scales[j]))
```

Reply:
I think that this assert is incorrect, but the example here does actually work because Normal broadcasts the tensors internally. In general, I think we need to either ensure that parameter tensors are broadcasted up to the same named dimensions or we will need a separate renaming dict for each parameter. I think broadcasting the parameters is simpler.

Regardless, I'm going to add some tests for this case.

Reply:
#255 refactors the distribution tests to consider cases like this.

Reply:
#255 revealed some bugs, which should now be fixed by broadcasting all distribution parameters before converting them to positional.